tests/publishing/templates.test.js

```diff
@@ -4,28 +4,34 @@ const common = require('../common'),
 
 describe('Templates', () => {
   before(() => common.login(browser, 'admin', 'admin', 'local'));
+  after(() => common.logout());
   it('should navigate to Templates', () => common.clickSidebarTab(browser, 'Templates'));
 
   it('should allow user to create a new template', () => {
-    let realHeader = '//div[@id="editTemplate"]//h1[@class="page-header"]';
-    return browser.click('//button[@id="new_tmpl"]')
-      .then(() => browser.waitForExist(realHeader))
-      //.then(() => browser.getSource().then(console.log))
-      .then(() => browser.getText(realHeader))
+    return common.waitForOnClick('//button[@id="new_tmpl"]')
+      .then(() => browser.click('//button[@id="new_tmpl"]'))
+      .then(() => browser.waitForExist('//div[@id="editTemplate"]//h1[@class="page-header"]'))
+      .then(() => browser.getText('//div[@id="editTemplate"]//h1[@class="page-header"]'))
       .then(title => expect(title).to.equal('Create New Template'));
   });
 
-  it('should fill in the new template data', () => {
-    let name = '//form[@id="newTmplForm"]//input[@name="name"]';
-    return browser.waitForEnabled(name)
-      .then(() => browser.setValue(name, 'Windows 10.1 x64 RE'))
-      .then(() => browser.getValue(name))
-      .then((text) => expect(text).to.equal('Windows 10.1 x64 RE'));
-    // .....
+  it('should allow user to cancel creating a new template', () => {
+    return common.waitForOnClick('//*[@id="newTmplForm"]/div[11]/button[2 and text()="Cancel"]')
+      .then(() => browser.click('//*[@id="newTmplForm"]/div[11]/button[2 and text()="Cancel"]'))
+      .then(() => browser.waitForExist('//*[@id="newTmplForm"]', 1000, true));
   });
 
+  it('should allow user to add a template', () => {
+    return browser.waitForExist('//*[@id="tmpl_form" and @class="hidden"]')
+      .then(() => common.waitForOnClick('//*[@id="add_tmpl"]'))
+      .then(() => browser.click('//*[@id="add_tmpl"]'))
+      .then(() => browser.waitForExist('//*[@id="add_tmpl_form"]'))
+  });
 
+  it('should allow user to cancel adding a template', () => {
+    return common.waitForOnClick('//*[@id="add_tmpl_form"]/div[3]/button[2 and text()="Cancel"]')
+      .then(() => browser.click('//*[@id="add_tmpl_form"]/div[3]/button[2 and text()="Cancel"]'))
+      .then(() => browser.waitForExist('//*[@id="tmpl_form" and @class="hidden"]'));
+  });
 
 });
-
```

Comment:
Should probably go under the tests/publishing/templates/ folder, and be called basic.test.js